charms/istio-pilot/src/charm.py

```diff
@@ -401,6 +412,39 @@ def _get_ingress_auth_data(self) -> dict:
 
         return ingress_auth_data
 
+    def _get_ingress_data(self, event) -> dict:
+        """Retrieve the ingress relation data without touching other interface data.
+
+        This is a workaround to ensure that errors in other relation data, such as an incomplete
+        ingress-auth relation, do not block us from retrieving the ingress data.
+        """
+        try:
+            ingress_interface = get_interface(self, INGRESS_RELATION_NAME)
+        except NoVersionsListed as err:
+            raise ErrorWithStatus(err, WaitingStatus)
+        except NoCompatibleVersions as err:
+            raise ErrorWithStatus(err, BlockedStatus)
+
+        # Filter out data we sent out.
+        if ingress_interface:
+            routes = {
+                (rel, app): route
+                for (rel, app), route in sorted(
+                    ingress_interface.get_data().items(), key=lambda tup: tup[0][0].id
+                )
+                if app != self.app
+            }
+        else:
+            # If there is no ingress-auth relation, we have no data here
+            routes = {}
+
+        if isinstance(event, RelationBrokenEvent):
+            # The app-level data is still visible on a broken relation, but we
+            # shouldn't be keeping the VirtualService for that related app.
+            del routes[(event.relation, event.app)]
```

Comment:
maybe `pop` the item?
have some cultural issues with del :D

Reply:
done!